src/_common/game/game.model.ts

```diff
@@ -516,3 +519,24 @@ export async function unfollowGame(game: Game) {
 		throw e;
 	}
 }
+
+export function handleGameAddFailure(user: User, reason: string, router: Router) {
+	switch (reason) {
+		case 'game-limit-reached':
+			showErrorGrowl(
+				$gettext(`You've reached the limit of how many games you can collaborate on.`)
+			);
+			router.push({
+				name: 'library.collection.developer',
+				params: { id: user.username },
+			});
+			break;
+
+		default:
+			showErrorGrowl($gettext(`You cannot add a new game or collaborate on this game.`));
+			router.push({
+				name: 'library.collection.developer',
+				params: { id: user.username },
+			});
+	}
+}
```

Comment:
We can probably just put the router push at the end of this function since we'll always call it.